src/System.Text.Json/src/System/Text/Json/Serialization/ReflectionEmitMaterializer.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if BUILDING_INBOX_LIBRARY
```

Comment:
What's causing us to only include this on .NET Core 3.0?

Reply:
Yes ideally we should add another define like "SUPPORTS_IL_EMIT" and have it defined in a single location probably with similar defines for the source code package.